File: niuniu-take-out-springboot3/server/src/main/java/common/niuniu/service/OrderService.java
package common.niuniu.service;

import common.niuniu.dto.*;
import common.niuniu.result.PageResult;
import common.niuniu.vo.OrderPaymentVO;
import common.niuniu.vo.OrderStatisticsVO;
import common.niuniu.vo.OrderSubmitVO;
import common.niuniu.vo.OrderVO;

public interface OrderService {
    OrderSubmitVO submit(OrderSubmitDTO orderSubmitDTO);

    Integer unPayOrderCount();

    OrderVO getById(Integer id);

    PageResult userPage(int page, int pageSize, Integer status);

    void userCancelById(Integer id) throws Exception;

    void reOrder(Integer id);

    OrderPaymentVO payment(OrderPaymentDTO orderPaymentDTO);

    PageResult conditionSearch(OrderPageDTO orderPageDTO);

    OrderStatisticsVO statistics();

    void confirm(OrderConfirmDTO orderConfirmDTO);

    void reject(OrderRejectionDTO orderRejectionDTO);

    void cancel(OrderCancelDTO orderCancelDTO);

    void delivery(Integer id);

    void complete(Integer id);

    void reminder(Integer id);
}

File: niuniu-take-out-kotlin/server/src/main/kotlin/common/niuniu/service/OrderService.kt
package common.niuniu.service

import common.niuniu.dto.*
import common.niuniu.result.PageResult
import common.niuniu.vo.OrderPaymentVO
import common.niuniu.vo.OrderStatisticsVO
import common.niuniu.vo.OrderSubmitVO
import common.niuniu.vo.OrderVO

interface OrderService {
    fun submit(orderSubmitDTO: OrderSubmitDTO?): OrderSubmitVO?

    fun unPayOrderCount(): Int?

    fun getById(id: Int?): OrderVO?

    fun userPage(page: Int, pageSize: Int, status: Int?): PageResult?

    @Throws(Exception::class)
    fun userCancelById(id: Int?)

    fun reOrder(id: Int?)

    fun payment(orderPaymentDTO: OrderPaymentDTO?): OrderPaymentVO?

    fun conditionSearch(orderPageDTO: OrderPageDTO?): PageResult?

    fun statistics(): OrderStatisticsVO?

    fun confirm(orderConfirmDTO: OrderConfirmDTO?)

    fun reject(orderRejectionDTO: OrderRejectionDTO?)

    fun cancel(orderCancelDTO: OrderCancelDTO?)

    fun delivery(id: Int?)

    fun complete(id: Int?)

    fun reminder(id: Int)
}
